lib/msf/core/db_manager/workspace.rb

```diff
@@ -21,102 +28,72 @@ def find_workspace(name)
-    ::Mdm::Workspace.order('updated_at asc').load
-  }
+    # Proxied to fix tests, will be cleaned up in remote test patch
+    framework.db.workspace=workspace
   end
 
-  #
-  # Returns an array of all the associated workspace records counts.
-  #
-  def workspace_associations_counts()
-    results = Array.new()
+  def workspaces(opts = {})
+  ::ActiveRecord::Base.connection_pool.with_connection {
+    search_term = opts.delete(:search_term)
+    # Passing these values to the search will cause exceptions, so remove them if they accidentally got passed in.
+    Msf::Util::DBManager.delete_opts_workspace(opts)
 
     ::ActiveRecord::Base.connection_pool.with_connection {
-      workspaces.each do |ws|
-        results << {
-            :name  => ws.name,
-            :hosts_count => ws.hosts.count,
-            :services_count => ws.services.count,
-            :vulns_count => ws.vulns.count,
-            :creds_count => ws.core_credentials.count,
-            :loots_count => ws.loots.count,
-            :notes_count => ws.notes.count
-        }
+      if search_term && !search_term.empty?
+        column_search_conditions = Msf::Util::DBManager.create_all_column_search_conditions(Mdm::Workspace, search_term)
+        Mdm::Workspace.where(opts).where(column_search_conditions)
+      else
+        Mdm::Workspace.where(opts)
       end
     }
-
-    return results
-  end
-
-  def delete_all_workspaces()
-    return delete_workspaces(workspaces.map(&:name))
+  }
   end
 
-  def delete_workspaces(names)
-    status_msg = []
-    error_msg = []
-
-    switched = false
-    # Delete workspaces
-    names.each do |name|
-      workspace = framework.db.find_workspace(name)
-      if workspace.nil?
-        error_msg << "Workspace not found: #{name}"
-      elsif workspace.default?
-        workspace.destroy
-        workspace = framework.db.add_workspace(name)
-        status_msg << 'Deleted and recreated the default workspace'
-      else
-        # switch to the default workspace if we're about to delete the current one
-        if framework.db.workspace.name == workspace.name
+  def delete_workspaces(opts)
+    raise ArgumentError.new("The following options are required: :ids") if opts[:ids].nil?
+    
+    ::ActiveRecord::Base.connection_pool.with_connection {
+      deleted = []
+      default_deleted = false
+      opts[:ids].each do |ws_id|
+        ws = Mdm::Workspace.find(ws_id)
+        default_deleted = true if ws.default?
+        if framework.db.workspace.name == ws.name
           framework.db.workspace = framework.db.default_workspace
-          switched = true
         end
-        # now destroy the named workspace
-        workspace.destroy
-        status_msg << "Deleted workspace: #{name}"
+        begin
+          deleted << ws.destroy
+          framework.db.workspace = framework.db.add_workspace(DEFAULT_WORKSPACE_NAME) if default_deleted
+        rescue
+          elog("Forcibly deleting #{workspace}")
```

Comment:
Why are we calling the `workspace` method for the value logged here? Should this be `ws.name` instead?